pkg/operator/ceph/cluster/mon/health.go

```diff
@@ -323,11 +323,12 @@ func (c *Cluster) handleExternalMonStatus(status client.MonStatusResponse) error
 
 func (c *Cluster) addOrRemoveExternalMonitor(status client.MonStatusResponse) (bool, error) {
 	var changed bool
-	// Populate a map with the current monitor from ClusterInfo
-	monsNotFound := map[string]interface{}{}
-	for _, mon := range c.ClusterInfo.Monitors {
-		monsNotFound[mon.Name] = struct{}{}
+	// clearing the content of clusterinfo monitors, it'll get populated in the next iteration
+	for mon := range c.ClusterInfo.Monitors {
+		delete(c.ClusterInfo.Monitors, mon)
 	}
+	logger.Debugf("ClusterInfo is now Empty, refilling it from status.MonMap.Mons")
+	monsNotFound := map[string]interface{}{}
```

Comment:
The iteration L378 becomes unnecessary now. We only need the iteration on `status.MonMap.Mons`.

Reply:
@leseb ok, I'll remove the iteration at L378 and that'll solve this issue.
wdyt

Reply:
It should. There is testing to after this.